utils/sanitize_dashboards.js

```diff
@@ -0,0 +1,46 @@
+const fs = require('fs');
+const { readdir } = require('fs').promises;
+// Removes the first two arguments passed by `node`, these aren't needed for this function to run - CM
+const pathsToSanitize = process.argv.slice(2);
+
+if (pathsToSanitize.length < 1)
+  console.log(
+    "You must pass a file path as an argument for the sanitize function to run \nExample: 'yarn sanitize-dashboard node-js/express'"
+  );
+
+const sanitizeDashboard = (fileContent) => {
+  return fileContent
+    .replace(/\"accountId\"\s*:\s*\d+\s*/g, '"accountId": 0') // Anonymize account ID
+    .replace(/^.+\"linkedEntityGuids\".+\n?/gm, '"linkedEntityGuids": null') // Remove linkedEntityGuids
+    .replace(/^.+\"permissions\".+\n?/gm, '') // Remove permissions
+    .replace(/[\}\)\]]\,(?!\s*?[\{\[\"\'\w])/g, '}'); // Remove trailing commas if any left after json blocks
+};
+
+pathsToSanitize.forEach(async (i) => {
+  const directory = `../quickstarts/${i}/dashboards/`;
+  const files = await readdir(directory, { withFileTypes: true });
```

Comment:
We can also use the [synchronous](https://nodejs.org/api/fs.html#fsreaddirsyncpath-options) version of this method if we don't want to mess with `async` loops.

Either way, we might want to do some error handling. I'm not exactly sure how this script would handle it if `readdir` throws an error (does it keep looping? does it terminate before everything is complete?)

Reply:
Considering that we use the synchronous version of `readFile` down below, I'd advocate for keeping everything synchronous. Since this is a one-off script, I don't think we care too much about it running in parallel.

Reply:
I tried to do some error handling just via `console.log`s, but they're not getting hit, the synchronous functions just error out, end the loop, and throw an error in the console like:
`$ node sanitize_dashboards.js nodejs/express
(node:14545) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, scandir '/Users/jmcadoo/projects/developer_enablement/newrelic-quickstarts/quickstarts/nodejs/express/dashboards'`

`fs.writeFile` can take a callback, so that was able to do some error handling if something goes awry, but the other two methods don't have callbacks as an option best i can tell.